testing/kuttl/e2e/pgbackrest-restore/07--update-cluster.yaml

```diff
@@ -0,0 +1,25 @@
+---
```

Comment:
💭 Do we know which of the 07 files will get run first? Is it always in alphabetic order (followed by the assert/error.yaml files)?

Reply:
`TestStep` commands always [run first](https://github.com/kudobuilder/kuttl/blob/d2cd0d5ec86a758ca4a13db032a89ae214d3d524/pkg/test/step.go#L411) (say [the docs](https://kuttl.dev/docs/testing/reference.html#teststep)).

Also, despite KEP 4, there can be only one `TestStep` per step.

Reply:
Ah, ok, as long as it's deterministic, I don't know that we need to separate them out with different numbers. 

(Also do people write teststeps that do deletes, then commands, then applies? I wonder when that would make sense/be more legible or maintainable vs. just separating it out over different index numbers.)

Reply:
> (Also do people write teststeps that do deletes, then commands, then applies? I wonder when that would make sense/be more legible or maintainable vs. just separating it out over different index numbers.)

I can't imagine they do. I suspect it's more that assertions should happen after everything else. "Everything else" gets lumped into one thing with a deterministic order.

I've wondered a bit about deletes. The default for new things (and definitely CRs) is [background deletion](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#cascading-deletion). Kuttl docs say it waits, but for what? Does it always use foreground deletion?